flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherTest.java

```diff
@@ -207,7 +207,13 @@ public void testJobSubmission() throws Exception {
 
     @Test
     public void testDuplicateJobSubmissionWithGloballyTerminatedJobId() throws Exception {
```

Comment:
We're still missing some tests:
* Test that the `JobResultStore` contains a clean `JobResultEntry` after everything went smoothly (happy path)
* We should test that the `JobResultEntry` remains in dirty state if any of the cleanup tasks fail